src/components/HomePage/HomePage.jsx

```diff
@@ -0,0 +1,124 @@
+/* eslint-disable import/extensions */
+import { useDispatch, useSelector } from 'react-redux';
+import './Login.css';
+import './HomePage.css';
+import React, { useEffect, useRef } from 'react';
+import { Row, Col } from 'react-bootstrap';
+import { useNavigate } from 'react-router-dom';
+import storage from '../../services/storageService';
+import { logoutUser } from '../../services/authService';
+import { carsSelector, fetchCarsDataAsync } from '../../redux/slices/cars';
+import Nav from '../Nav';
+
+function HomePage() {
+  const history = useNavigate();
+  const dispatch = useDispatch();
+  const mainDiv = useRef();
+  const carArray = useSelector((state) => carsSelector(state));
+  const handleLogOut = () => {
+    dispatch(logoutUser(history));
+    history('/login');
+  };
+  const goToDetailsPage = (carId) => {
+    history('/details', { state: { carId } });
+  };
+  const cars = carArray.map((car) => (
+    <Col
+      key={car.id}
+      sm={12}
+      md={4}
+      onClick={() => goToDetailsPage(car.id)}
+    >
+      <div className="d-flex flex-column justify-content-center align-items-center">
+        <img src={car.image} alt="" className="car-img" />
+        <h5 className="mt-3 fsize-15">{car.name.toUpperCase()}</h5>
+        <div className="d-flex fsize-3 mb-4 text-muted">
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
+          <i className="fa fa-circle me-1" />
```

Comment:
- [ ] Maybe this piece of HTML could be inside of a function that returns it since you are using it twice in this file? And a loop might be a better option too, this way your code would be cleaner.

Reply:
Issued fixed!